Dockerfile

```diff
@@ -0,0 +1,21 @@
+FROM golang:1.6-alpine
+
+RUN mkdir -p /go/src/redoctober
+WORKDIR /go/src/redoctober
+
+# https://github.com/gliderlabs/docker-alpine/blob/master/docs/usage.md#disabling-cache
+RUN apk --no-cache add git openssl
+
+# build binary
+COPY . /go/src/redoctober
+RUN go get -d -v
+RUN go install -v
+
+# cleanup
+RUN apk del git
+RUN rm -rf /var/cache/apk/*
```

Comment:
Yes for sure it does optimize re-build times, depending on where the cache
is busted, but since this will ideally only be built on pushes to master it
isn't that big a deal, _if_ instead they were using this image to test
redoctober, then I would take a whole different approach :)

On Tue, Mar 15, 2016 at 1:19 PM, Aidan Feldman <notifications@github.com>
wrote:

> In Dockerfile
> <https://github.com/cloudflare/redoctober/pull/131#discussion_r56233386>:
>
> > +FROM golang:1.6-alpine
> > +
> > +RUN mkdir -p /go/src/redoctober
> > +WORKDIR /go/src/redoctober
> > +
> > +# https://github.com/gliderlabs/docker-alpine/blob/master/docs/usage.md#disabling-cache
> > +RUN apk --no-cache add git openssl
> > +
> > +# build binary
> > +COPY . /go/src/redoctober
> > +RUN go get -d -v
> > +RUN go install -v
> > +
> > +# cleanup
> > +RUN apk del git
> > +RUN rm -rf /var/cache/apk/*
>
> Would it be fair to say that having multiple RUNs optimizes re-build
> times, whereas having them all combined into one RUN optimizes the
> download size of the image? Totally fine with switching it—just want to
> understand the rationale.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/cloudflare/redoctober/pull/131/files/170f1e4c10820682d977ed30aefa96a7e3c3cdc4#r56233386>
>



-- 


Jessie Frazelle
4096R / D4C4 DD60 0D66 F65A 8EFC  511E 18F3 685C 0022 BFF3
pgp.mit.edu <http://pgp.mit.edu/pks/lookup?op=get&search=0x18F3685C0022BFF3>